src/controllers/keystore/keystore.ts

```diff
@@ -175,16 +199,21 @@ export class KeystoreController extends EventEmitter {
     const aesCtr = new aes.ModeOfOperation.ctr(derivedKey, counter)
     const mac = keccak256(concat([macPrefix, aesEncrypted.ciphertext]))
     if (mac !== aesEncrypted.mac) {
-      // Throw, because that's handled as a form field error
-      throw new Error('keystore: wrong secret')
+      this.errorMessage = 'Invalid Device Password.'
+      throw new EmittableError({
```

Comment:
here we need only the form-level error and not the error toast so omit emitting an error and emit only an update